CondCore/CondDB/interface/IOVProxy.h

```diff
@@ -104,7 +104,8 @@ namespace cond {
       // the only way to construct it from scratch...
       explicit IOVProxy(const std::shared_ptr<SessionImpl>& session);
 
-      //
+      // TODO This seems to violate The rule of three: why copy constructor and assignment is defined but not the descructor?
+      // Is this ctor even needed? How is it different from the implicitly defined one?
```

Comment:
should this become an issue instead of a comment?

Reply:
It would be better to refer to the rule of five (i.e. including the move constructor and assignment as well). By quick look all of the four could just be defaulted (which also means they could be left to be implicitly declared by the constructor).

I think it would be good for this PR to do something for them, as in the present state of the PR the `m_printDebug` does not get forwarded through the copy operations.

Reply:
The default ctor and copy ctor that were implemented there were not different from what would be implicitly generated (until I added the `m_printDebug` member, thank you for a very good catch @makortel!)   

* I changed the explicit implementation of default ctor to `= default`
   * some doubts described later
* removed the explicit copy ctor, now it should be implicitly generated
* removed the explitic copy assign operator,  now it should be implicitly generated
* move assign and move ctor should be implicitly generated

Let me know if I made any mistakes.

The ctor from a session is commented with `// the only way to construct it from scratch...` - does it mean the default constructor should be removed?... I doubt it, but its a bit confusing